finetune/base_models/gpt/featurizer.py

```diff
@@ -174,10 +178,22 @@ def gpt_featurizer(X, encoder, config, train=False, reuse=None):
                                              scope='h%d' % layer, train=train_layer, scale=True)
                 if config.low_memory_mode and train_layer:
                     block_fn = recompute_grad(block_fn, use_entire_scope=True)
-                h = block_fn(h)
+                if layer < config.n_layer - 1:
+                    h = block_fn(h)
+                else:
+                    h_out = block_fn(h)
+
+            # get the attention weights from the last layer
+            if layer == config.n_layer - 1:
+                with tf.variable_scope('h%d_' % layer, reuse=True):
```

Comment:
Yup like Ben said we can just use slashes here to prevent having to do this waterfall thing